packages/compass-workspaces/src/services/workspaces-storage.tsx

```diff
@@ -0,0 +1,102 @@
+import { createServiceLocator } from '@mongodb-js/compass-app-registry';
+import {
+  IUserData,
+  type ReadAllResult,
+  z,
+} from '@mongodb-js/compass-user-data';
+import React, { useContext, useRef } from 'react';
+
+/**
+ * Schema for saving workspace tab state to user data
+ */
+
+// Define schema for collection subtab
```

Comment:
Almost everything here is exact copy of types defined in `packages/compass-workspaces/src/types.ts`, to avoid these diverging we should use one of those as source. As you'd need the schemas to use with UserData, we should probalby make schemas the source of types. This will require you to split the schemas here a bit to follow the type structure that we need in the app